### PR TITLE
Import  flow_trigger_discovery_webhook and migrate to flow trigger callback

### PR DESCRIPTION
**Pull request by mikee649**
### WHY are these changes introduced?

Closes https://github.com/Shopify/flow/issues/27654

### WHAT is this pull request doing?

- Allows the cli to import `flow_trigger_discovery_webhook` and migrate it to a `flow_trigger_lifecycle_callback`

### How to test your changes?

See the 🎩 ing described [here](https://github.com/Shopify/flow/issues/27830#issuecomment-2444954241)

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

File: packages/app/src/cli/services/flow/extension-to-toml.ts [hunk 1]
```diff
@@ -35,20 +35,36 @@ export function buildTomlObject(extension: ExtensionRegistration): string {
...
+    localExtensionRepresentation = {
+      extensions: [
+        {
+          type: 'flow_trigger_lifecycle_callback',
+          name: extension.title,
+          handle: slugify(extension.title.substring(0, MAX_EXTENSION_HANDLE_LENGTH)),
+          url: config.url ?? defaultURL,
+        },
+      ],
+      settings: (fields?.length ?? 0) > 0 ? {fields} : undefined,
```
[the remark this answers is not in the document]
Comment (null-sleep) [reply packages/app/src/cli/services/flow/extension-to-toml.ts:50]: Fair enough, easy enough to remove

[the remark this answers is not in the document]
Comment (null-sleep) [reply packages/app/src/cli/services/flow/extension-to-toml.ts:50]: > Is this why description above was made nullable?
Yes

[the remark this answers is not in the document]
Comment (null-sleep) [reply packages/app/src/cli/services/flow/extension-to-toml.ts:50]: > I wonder if this should be more clearly separated from actions and triggers

I think its easy enough to make a new `interface FlowConfig` and name them to be clearer

**Comment by null-sleep** [comment]
tested again on spin:
```
[[extensions]]
type = "flow_trigger_lifecycle_callback"
name = "old webhook"
handle = "old-webhook"
url = "https://shopify.com"
```
The ext file was correctly created

File: packages/app/src/cli/services/flow/extension-to-toml.ts [hunk 2]
```diff
@@ -22,33 +23,53 @@ interface FlowConfig {
...
+
+  if (extension.type === 'flow_trigger_discovery_webhook') {
+    const config: FlowWebhookConfig = JSON.parse(versionConfig)
+    localExtensionRepresentation = {
+      extensions: [
+        {
+          type: 'flow_trigger_lifecycle_callback',
+          name: extension.title,
+          handle: slugify(extension.title.substring(0, MAX_EXTENSION_HANDLE_LENGTH)),
+          url: config.url ?? defaultURL,
```
[the remark this answers is not in the document]
Comment (null-sleep) [reply packages/app/src/cli/services/flow/extension-to-toml.ts]: Actually we dont need `defaultURL` there, it should just be `config.url`. Removed

**Review by isaacroldan — APPROVED** [review]

**Comment by mikee649** [comment]
> I went ahead and made the update for you, please let me know if something doesn't work as it should (but the PR is good to be merged)

Thanks @isaacroldan, 🎩 LGTM, I'm going to go ahead and merge

**Pull request merged**